src/main/java/com/box/sdk/BoxFile.java

```diff
@@ -604,6 +607,60 @@ public Metadata createMetadata(String typeName, Metadata metadata) {
     }
 
     /**
+     * Locks a file
+     * @param expiresAt expiration date of the lock
+     * @return the lock returned from the server
+     */
+    public BoxLock lock(Date expiresAt) {
+        return this.lock(expiresAt, false);
+    }
+
+    /**
+     * Locks a file
+     * @param expiresAt expiration date of the lock
+     * @param isDownloadPrevented is downloading of file prevented when locked
+     * @return the lock returned from the server
+     */
+    public BoxLock lock(Date expiresAt, boolean isDownloadPrevented) {
+        String queryString = new QueryStringBuilder().appendParam("fields", "lock").toString();
+        URL url = FILE_URL_TEMPLATE.buildWithQuery(this.getAPI().getBaseURL(), queryString, this.getID());
+        BoxAPIRequest request = new BoxAPIRequest(this.getAPI(), url, "PUT");
+        request.addHeader("Content-Type", "application/json");
+
+        JsonObject lockConfig = new JsonObject();
+        lockConfig.add("type", "lock");
+        lockConfig.add("expires_at", BoxDateFormat.format(expiresAt));
+        lockConfig.add("is_download_prevented", Boolean.toString(isDownloadPrevented));
+
+        JsonObject requestJSON = new JsonObject();
+        requestJSON.add("lock", lockConfig);
+        request.setBody(requestJSON.toString());
+
+        BoxJSONResponse response = (BoxJSONResponse)request.send();
+
+        JsonObject responseJSON = JsonObject.readFrom(response.getJSON());
+        JsonValue lockValue = responseJSON.get("lock");
+        JsonObject lockJSON = JsonObject.readFrom(lockValue.toString());
+
+        return new BoxLock(lockJSON, this.getAPI());
```

Comment:
I noticed that we sometimes do a `response.disconnect()`, should I add that to these functions?